### DISCUSSION TITLE
Anyone having problems with php mail() function turned off?

### DISCUSSION THREAD

**Discussion by pawell0** [datenstrom/community#865] [Ask a question] [answered]
Please don't laugh at me that loud. 😆 I had a little stupid situation with my website based on Yellow (latest). Previous year my hosting provider turned off the possibility to use the php mail() function. Only possibility to send emails from any of my websites is to use the smtp (using phpmailer or similar scripts). A few days ago I had to reset the pasword to my Yellow based website. I tried to do this directly using the website. You probably know what was next ... Yellow could not send emails from my web hosting so I could not reset my password. Anyway I started the process of  resetting the password and I could not login using my old password anymore. At last I had to reinstall the website from scratch, keeping the data some way because I did not find any other way to solve the problem. Additionally the Contact plugin stopped to work ... 
I leave this history here for your consideration because I'm possibly not last user with that kind of problems.
Anyway - any advise and ideas to solve those problems in the future are welcome ... Or maybe someone already wrote something on this but I have not read it?
Yet another observation: Yellow is relativelly easy to install, setup and maintain even by the ordinary computer user with limited skills, and has the great documentation. Big thanks for that! But it's always worth to see the matter from the ordinary user's point of view and keep things easy. Without any programming knowledge ordinary user will never know why suddenly he can't reset a password or manage users using website or send emails by forms. And how to solve these problems by himself. Thank you in advance!

**Comment by wunderfeyd** [comment, marked as answer]
Maybe a little bit off topic but...

hints from my side to restore access to a yellow system (if mail is somehow broken):

* Use the command line (if available)

  `php yellow.php user change <email> <password>`

  or

  `php yellow.php user add <email> <password>`

* Update your `system/extensions/yellow-user.ini` manually

  Search for your administration account and replace the "Hash" value with the output of a bcrypt hash generator like https://www.coderstool.com/bcrypt-hash-generator.

**Reply by nevillepark** [reply]
Hi, I'm having a similar problem (I want to modify user details, but mail won't work). I want to edit `yellow-user.ini`. What _exactly_ do I need to hash?

**Reply by wunderfeyd** [reply]
Simply your password is needed for hashing. 

`php yellow.php user change <email> <password>` does the same for you if the password has to be modified only.

**Reply by nevillepark** [reply]
Thanks. Oddly, the hash in the "Hash: " field doesn't correspond to my password. It is also different from the hash that comes after the pending email address. Is it supposed to be like that?

**Reply by annaesvensson** [reply]
Yes, encrypted passwords are stored instead of plaintext passwords. That's why we included tools to change passwords in a web browser and at the command line. See also https://github.com/annaesvensson/yellow-edit

**Reply by nevillepark** [reply]
Oh, no, I meant that *the computed hash* for my password was not the same as the one in yellow-user.ini.

**Reply by wunderfeyd** [reply]
The computed hash includes a randomized salt. In this case a same password will lead to a different output when creating the hash. You can't compare the hashes directly. This is indeed a builtin security feature to make it harder for an attacker to find accounts with the same password if the user database got into the wrong hands somehow.

**Reply by nevillepark** [reply]
Oh, I see! Thank you.
